src/main/java/ca/on/oicr/pineryreports/util/SampleUtils.java

```diff
@@ -115,6 +115,16 @@ public static SampleDto getParent(SampleDto sample, Map<String, SampleDto> poten
     return parent;
   }
   
+  public static SampleDto getParent(SampleDto sample, String sampleCategory, Map<String, SampleDto> potentialParents) {
+    for (SampleDto current = sample; current != null; current = getParent(current, potentialParents)) {
+      if (sampleCategory.equals(getAttribute("Sample Category", current))) {
+        return current;
+      }
+    }
+    throw new IllegalStateException("Parent " + sampleCategory + " of sample " + sample.getId()
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "NullPointerException" could be thrown; "sample" is nullable here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2259)